### PR TITLE
Adding caching for HostAddressManager

### PR DESCRIPTION
**Pull request by jayaramcs**
- addUsage
- getHostAddress

### CODE REVIEW

File: bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java [hunk 1]
```diff
@@ -52,12 +52,19 @@
...
 	/**
-	 * Recently added or accessed Host Address Object Ids are cached. This is
+	 * Recently added or accessed Host Address Objects are cached. This is
 	 * here to improve performance of the
 	 * {@link #hostAddressExists(org.sleuthkit.datamodel.HostAddress.HostAddressType, java.lang.String)}
-	 * check.
+	 * check as well as the {@link #getHostAddress(org.sleuthkit.datamodel.HostAddress.HostAddressType, java.lang.String) }
 	 */
-	private final Cache<String, Long> recentHostAddresstCache = CacheBuilder.newBuilder().maximumSize(200000).build();
+	private final Cache<String, HostAddress> recentHostAddresstCache = CacheBuilder.newBuilder().maximumSize(200000).build();
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java:60]: Can you fix the existing typo here Addresst->Address

File: bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java [hunk 2]
```diff
@@ -98,6 +105,12 @@
 	 * @throws TskCoreException
 	 */
 	private Optional<HostAddress> getHostAddress(HostAddress.HostAddressType type, String address, CaseDbConnection connection) throws TskCoreException {
+		
+		HostAddress hostAddress = recentHostAddresstCache.getIfPresent(type.getId() + "#" + address.toLowerCase());
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java:109]: Can we put the "type.getId() + "#" + address.toLowerCase()" part into a method like "createRecentHostAddressKey()"?

File: bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java [hunk 3]
```diff
@@ -587,13 +604,21 @@ public boolean hostNameAndIpMappingExists(long addressObjectId) throws TskCoreEx
 	 * @param hostAddress The host address.
 	 */
 	public void addUsage(Content content, HostAddress hostAddress) throws TskCoreException {
+		
+		String key = content.getDataSource().getId() + "#" + hostAddress.getId();
+		Long cachedValue = hostAddressUsageCache.getIfPresent(key);
+		if (null != cachedValue && cachedValue == content.getId()) {
```
Comment (APriestman) [inline bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java:610]: Is there any reason to do the second part of this check?

Comment (jayaramcs) [reply bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java:610]: @APriestman Yes; usage is associated with both a data source as well as the content. The key is on datasourceId.

Comment (APriestman) [reply bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java:610]: Ok I see that now, but then why wouldn't the key be on all three fields? Isn't the point just to check if you've already added (hostAddrId, contentId, datasourceId) to the database?

Comment (jayaramcs) [reply bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java]: I have updated it with all 3 now

File: bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java (hunk 1 above)
Comment (jayaramcs) [reply bindings/java/src/org/sleuthkit/datamodel/HostAddressManager.java]: @APriestman changes are in

**Review by APriestman — APPROVED** [review]

**Pull request merged**